features/lwipstack/lwip/src/core/ipv4/lwip_autoip.c

```diff
@@ -94,6 +94,9 @@
 static err_t autoip_arp_announce(struct netif *netif);
 static void autoip_start_probing(struct netif *netif);
 
+/* static variables */
+static u8_t is_autoip_alloced = 0;      /* Set when 'struct autoip' is allocated dynamically and assigned to 'netif' */
```

Comment:
Why is this needed? Isn't `if (autoip == NULL)` enough?

Reply:
LwIP provides a way for the caller to assign the static autoip client (structure). This can be achieved by calling **_autoip_set_struct()_** API. When the caller allocates the autoip client and assigns it to the netif interface by calling **_autoip_set_struct()_** API, LwIP doesn't allocate the autoip client in **_autoip_start()_** API. Hence, if the autoip client (structure) is allocated by the caller and assigned it to netif interface by calling **_autoip_set_struct()_** API, then autoip structure shouldn't be freed in **_autoip_stop()_**. This flag is used to differentiate the above case.

Reply:
Ok, thanks for the answer, could you fix the naming to ..._allocated

Reply:
Sure

Reply:
@AnttiKauppila - Renamed the flag. Committed in SHA#a4aeee941d263e0e5d05c561ca6689e581af214b